diff --git a/base/files/file_util.h b/base/files/file_util.h
index 8f7d53c..81b8e8c 100644
--- a/base/files/file_util.h
+++ b/base/files/file_util.h
@@ -220,6 +220,11 @@ BASE_EXPORT bool GetTempDir(FilePath* path);
 // Path service may also override DIR_HOME.
 BASE_EXPORT FilePath GetHomeDir();
 
+// Get the caches directory. Basically is used only for OS X.
+//
+// At the moment returns 'false' on any other system than Mac OS.
+BASE_EXPORT bool GetCachesDir(FilePath* path);
+
 // Creates a temporary file. The full path is placed in |path|, and the
 // function returns true if was successful in creating the file. The file will
 // be empty and all handles closed after this function returns.
diff --git a/base/files/file_util_mac.mm b/base/files/file_util_mac.mm
index e9c6c65..e1cc741 100644
--- a/base/files/file_util_mac.mm
+++ b/base/files/file_util_mac.mm
@@ -23,11 +23,41 @@ bool CopyFile(const FilePath& from_path, const FilePath& to_path) {
 }
 
 bool GetTempDir(base::FilePath* path) {
+#ifndef MAC_APP_STORE
   NSString* tmp = NSTemporaryDirectory();
   if (tmp == nil)
     return false;
   *path = base::mac::NSStringToFilePath(tmp);
   return true;
+#else // MAC_APP_STORE
+  // Chromium uses temp directories to store a lot of UNIX socket files.
+  // In sandboxed applications socket files can only be used if they are
+  // inside of app container -
+  // https://stackoverflow.com/questions/17753222/sandboxd-deny-network-bind-error-message-when-binds-socket
+  // That is why it is easier to put the whole tmp dir on mac
+  // in app container rather than fix it all over the places.
+  base::FilePath cachesDir;
+  if (!GetCachesDir(&cachesDir)) {
+    return false;
+  }
+  cachesDir = cachesDir.Append("tmp");
+  if (!DirectoryExists(cachesDir)) {
+    if (mkdir(cachesDir.value().c_str(), 0700) != 0) {
+      return false;
+    }
+  }
+  *path = cachesDir;
+  return true;
+#endif
+}
+
+bool GetCachesDir(base::FilePath* path) {
+  NSArray *paths = NSSearchPathForDirectoriesInDomains(NSCachesDirectory, NSUserDomainMask, YES);
+  NSString *cachesDirectory = [paths objectAtIndex:0];
+  if (cachesDirectory == nil)
+    return false;
+  *path = base::mac::NSStringToFilePath(cachesDirectory);
+  return true;
 }
 
 FilePath GetHomeDir() {
diff --git a/base/files/file_util_posix.cc b/base/files/file_util_posix.cc
index ca1c5250..3621fdc 100644
--- a/base/files/file_util_posix.cc
+++ b/base/files/file_util_posix.cc
@@ -137,7 +137,7 @@ std::string TempFileName() {
 #if defined(GOOGLE_CHROME_BUILD)
   return std::string(".com.google.Chrome.XXXXXX");
 #else
-  return std::string(".org.chromium.Chromium.XXXXXX");
+  return std::string(".de.iridiumbrowser.XXXXXX");
 #endif
 }
 
diff --git a/base/mac/foundation_util.mm b/base/mac/foundation_util.mm
index 6ae5df3..5daf17c 100644
--- a/base/mac/foundation_util.mm
+++ b/base/mac/foundation_util.mm
@@ -242,7 +242,7 @@ const char* BaseBundleID() {
 #if defined(GOOGLE_CHROME_BUILD)
   return "com.google.Chrome";
 #else
-  return "org.chromium.Chromium";
+  return "de.iridiumbrowser";
 #endif
 }
 
diff --git a/base/mac/mach_port_broker.mm b/base/mac/mach_port_broker.mm
index bd47017..e90f1cf 100644
--- a/base/mac/mach_port_broker.mm
+++ b/base/mac/mach_port_broker.mm
@@ -73,8 +73,14 @@ std::string MachPortBroker::GetMachPortName(const std::string& name,
                                             bool is_child) {
   // In child processes, use the parent's pid.
   const pid_t pid = is_child ? getppid() : getpid();
+#ifndef MAC_APP_STORE
   return base::StringPrintf(
       "%s.%s.%d", base::mac::BaseBundleID(), name.c_str(), pid);
+#else
+  // @APPLE_TEAM_ID@ should be set before building and should be the same as in signing script!!!
+  // We rely on it to be correct in order to setup mach based IPC in sandboxed environment.
+  return base::StringPrintf("@APPLE_TEAM_ID at .%s.rohitfork.%d", base::mac::BaseBundleID(), pid);
+#endif
 }
 
 mach_port_t MachPortBroker::TaskForPid(base::ProcessHandle pid) const {
diff --git a/base/memory/shared_memory_posix.cc b/base/memory/shared_memory_posix.cc
index d55c2df..72a6e2f 100644
--- a/base/memory/shared_memory_posix.cc
+++ b/base/memory/shared_memory_posix.cc
@@ -455,7 +455,7 @@ bool SharedMemory::FilePathForMemoryName(const std::string& mem_name,
 #if defined(GOOGLE_CHROME_BUILD)
   std::string name_base = std::string("com.google.Chrome");
 #else
-  std::string name_base = std::string("org.chromium.Chromium");
+  std::string name_base = std::string("de.iridiumbrowser");
 #endif
   *path = temp_dir.AppendASCII(name_base + ".shmem." + mem_name);
   return true;
diff --git a/build/chrome_settings.gypi b/build/chrome_settings.gypi
index 646b209..f7ee963 100644
--- a/build/chrome_settings.gypi
+++ b/build/chrome_settings.gypi
@@ -20,10 +20,10 @@
             # The policy .grd file also needs the bundle id.
             'grit_defines': ['-D', 'mac_bundle_id=com.google.Chrome'],
           }, {  # else: branding!="Chrome"
-            'mac_bundle_id': 'org.chromium.Chromium',
+            'mac_bundle_id': 'de.iridiumbrowser',
             'mac_creator': 'Cr24',
             # The policy .grd file also needs the bundle id.
-            'grit_defines': ['-D', 'mac_bundle_id=org.chromium.Chromium'],
+            'grit_defines': ['-D', 'mac_bundle_id=de.iridiumbrowser'],
           }],  # branding
         ],  # conditions
       }],  # OS=="mac"
diff --git a/build/mac/tweak_info_plist.py b/build/mac/tweak_info_plist.py
index a6ffbca..6a7b358 100755
--- a/build/mac/tweak_info_plist.py
+++ b/build/mac/tweak_info_plist.py
@@ -80,7 +80,7 @@ def _AddVersionKeys(plist, version=None):
     VERSION_FILE = os.path.join(TOP, 'chrome/VERSION')
 
     (stdout, retval1) = _GetOutput([VERSION_TOOL, '-f', VERSION_FILE, '-t',
-                                    '@MAJOR@.@MINOR@.@BUILD@.@PATCH@'])
+                                    '@MAJOR@.@MINOR@.@BUILD@'])
     full_version = stdout.rstrip()
 
     (stdout, retval2) = _GetOutput([VERSION_TOOL, '-f', VERSION_FILE, '-t',
@@ -102,7 +102,7 @@ def _AddVersionKeys(plist, version=None):
   # http://lists.apple.com/archives/carbon-dev/2006/Jun/msg00139.html
   # BUILD will always be an increasing value, so BUILD_PATH gives us something
   # unique that meetings what LS wants.
-  plist['CFBundleVersion'] = bundle_version
+  plist['CFBundleVersion'] = full_version
 
   # Return with no error.
   return True
diff --git a/chrome/app/app-Info.plist b/chrome/app/app-Info.plist
index c5939d4..95d5f46 100644
--- a/chrome/app/app-Info.plist
+++ b/chrome/app/app-Info.plist
@@ -244,6 +244,8 @@
 			<string>Viewer</string>
 		</dict>
 		<dict>
+        <key>CFBundleTypeName</key>
+        <string>Extension viewer</string>
 			<key>CFBundleTypeRole</key>
 			<string>Viewer</string>
 			<key>LSItemContentTypes</key>
@@ -348,6 +350,8 @@
 			</dict>
 		</dict>
 	</array>
+ <key>LSApplicationCategoryType</key>
+ <string>public.app-category.utilities</string>
         <key>NSUserNotificationAlertStyle</key>
         <string>banner</string>
 </dict>
diff --git a/chrome/app_shim/app_shim.gypi b/chrome/app_shim/app_shim.gypi
index 242b660..550cb61 100644
--- a/chrome/app_shim/app_shim.gypi
+++ b/chrome/app_shim/app_shim.gypi
@@ -63,7 +63,10 @@
       ],
       'xcode_settings': {
         'INFOPLIST_FILE': 'app_shim/app_mode-Info.plist',
-        'APP_MODE_APP_BUNDLE_ID': '<(mac_bundle_id).app.@APP_MODE_SHORTCUT_ID@',
+        # There is no other mentions of @APP_MODE_SHORTCUT_ID@ for substitution.
+        # Temporary replace @APP_MODE_SHORTCUT_ID@ in next line with correct bundleID
+        #'APP_MODE_APP_BUNDLE_ID': '<(mac_bundle_id).app.@APP_MODE_SHORTCUT_ID@',
+        'APP_MODE_APP_BUNDLE_ID': '<(mac_bundle_id).app.app-mode-loader',
       },
       'postbuilds' : [
         {
diff --git a/chrome/browser/app_controller_mac.mm b/chrome/browser/app_controller_mac.mm
index 5e46b72..fe894c9 100644
--- a/chrome/browser/app_controller_mac.mm
+++ b/chrome/browser/app_controller_mac.mm
@@ -164,7 +164,7 @@ CFStringRef BaseBundleID_CFString() {
   return base::mac::NSToCFCast(base_bundle_id);
 }
 
-// This callback synchronizes preferences (under "org.chromium.Chromium" or
+// This callback synchronizes preferences (under "de.iridiumbrowser" or
 // "com.google.Chrome"), in particular, writes them out to disk.
 void PrefsSyncCallback() {
   if (!CFPreferencesAppSynchronize(BaseBundleID_CFString()))
diff --git a/chrome/browser/apps/app_shim/app_shim_host_manager_browsertest_mac.mm b/chrome/browser/apps/app_shim/app_shim_host_manager_browsertest_mac.mm
index 38e9cb6..91bd6ea 100644
--- a/chrome/browser/apps/app_shim/app_shim_host_manager_browsertest_mac.mm
+++ b/chrome/browser/apps/app_shim/app_shim_host_manager_browsertest_mac.mm
@@ -243,7 +243,7 @@ bool AppShimHostManagerBrowserTestSocketFiles::SetUpUserDataDirectory() {
   symlink_path_ = user_data_dir.Append(app_mode::kAppShimSocketSymlinkName);
   base::FilePath temp_dir;
   PathService::Get(base::DIR_TEMP, &temp_dir);
-  EXPECT_TRUE(base::CreateSymbolicLink(temp_dir.Append("chrome-XXXXXX"),
+  EXPECT_TRUE(base::CreateSymbolicLink(temp_dir.Append("ir-XXXXXX"),
                                        symlink_path_));
 
   // Create an invalid RunningChromeVersion file.
diff --git a/chrome/browser/apps/app_shim/app_shim_host_manager_mac.mm b/chrome/browser/apps/app_shim/app_shim_host_manager_mac.mm
index c7f1f86..e4e548c 100644
--- a/chrome/browser/apps/app_shim/app_shim_host_manager_mac.mm
+++ b/chrome/browser/apps/app_shim/app_shim_host_manager_mac.mm
@@ -37,9 +37,9 @@ base::FilePath GetDirectoryInTmpTemplate(const base::FilePath& user_data_dir) {
   base::FilePath temp_dir;
   CHECK(PathService::Get(base::DIR_TEMP, &temp_dir));
   // Check that it's shorter than the IPC socket length (104) minus the
-  // intermediate folder ("/chrome-XXXXXX/") and kAppShimSocketShortName.
-  DCHECK_GT(83u, temp_dir.value().length());
-  return temp_dir.Append("chrome-XXXXXX");
+  // intermediate folder ("/ir-XXXXXX/") and kAppShimSocketShortName.
+  DCHECK_GT(87u, temp_dir.value().length());
+  return temp_dir.Append("ir-XXXXXX");
 }
 
 void DeleteSocketFiles(const base::FilePath& directory_in_tmp,
diff --git a/chrome/browser/prefs/pref_functional_browsertest.cc b/chrome/browser/prefs/pref_functional_browsertest.cc
index b449b607..5be4685 100644
--- a/chrome/browser/prefs/pref_functional_browsertest.cc
+++ b/chrome/browser/prefs/pref_functional_browsertest.cc
@@ -203,7 +203,7 @@ IN_PROC_BROWSER_TEST_F(PrefsFunctionalTest, TestHomepagePrefs) {
 IN_PROC_BROWSER_TEST_F(PrefsFunctionalTest, PRE_TestPrivacySecurityPrefs) {
   PrefService* prefs = browser()->profile()->GetPrefs();
 
-  static_assert(chrome_browser_net::NETWORK_PREDICTION_DEFAULT !=
+/*  static_assert(chrome_browser_net::NETWORK_PREDICTION_DEFAULT !=
                     chrome_browser_net::NETWORK_PREDICTION_NEVER,
                 "PrefsFunctionalTest.TestPrivacySecurityPrefs relies on "
                 "predictive network actions being enabled by default.");
@@ -211,7 +211,7 @@ IN_PROC_BROWSER_TEST_F(PrefsFunctionalTest, PRE_TestPrivacySecurityPrefs) {
             prefs->GetInteger(prefs::kNetworkPredictionOptions));
   prefs->SetInteger(prefs::kNetworkPredictionOptions,
                     chrome_browser_net::NETWORK_PREDICTION_NEVER);
-
+*/
   EXPECT_TRUE(prefs->GetBoolean(prefs::kSafeBrowsingEnabled));
   prefs->SetBoolean(prefs::kSafeBrowsingEnabled, false);
 
diff --git a/chrome/browser/process_singleton_posix.cc b/chrome/browser/process_singleton_posix.cc
index d5394b8..1976082 100644
--- a/chrome/browser/process_singleton_posix.cc
+++ b/chrome/browser/process_singleton_posix.cc
@@ -733,7 +733,23 @@ ProcessSingleton::ProcessSingleton(
     : notification_callback_(notification_callback),
       current_pid_(base::GetCurrentProcId()),
       watcher_(new LinuxWatcher(this)) {
+#if defined(OS_MACOSX)
+#ifndef MAC_APP_STORE
   socket_path_ = user_data_dir.Append(chrome::kSingletonSocketFilename);
+#else // not def MAC_APP_STORE
+  // We do this to mitigate the problem with 104 chars max socket path in OSX
+  bool gotCachesDir = base::GetCachesDir(&socket_path_);
+  if (gotCachesDir) {
+    socket_path_ = socket_path_.Append(chrome::kSingletonSocketFilename);
+  } else {
+    socket_path_ = user_data_dir.Append(chrome::kSingletonSocketFilename);
+  }
+#endif // not def MAC_APP_STORE
+
+#else // defined(OS_MACOSX)
+  socket_path_ = user_data_dir.Append(chrome::kSingletonSocketFilename);
+#endif // defined(OS_MACOSX)
+
   lock_path_ = user_data_dir.Append(chrome::kSingletonLockFilename);
   cookie_path_ = user_data_dir.Append(chrome::kSingletonCookieFilename);
 
@@ -974,14 +990,21 @@ bool ProcessSingleton::Create() {
       << "Temp directory mode is not 700: " << std::oct << dir_mode;
 
   // Setup the socket symlink and the two cookies.
+#ifndef MAC_APP_STORE
   base::FilePath socket_target_path =
       socket_dir_.path().Append(chrome::kSingletonSocketFilename);
+#endif
   base::FilePath cookie(GenerateCookie());
   base::FilePath remote_cookie_path =
       socket_dir_.path().Append(chrome::kSingletonCookieFilename);
   UnlinkPath(socket_path_);
   UnlinkPath(cookie_path_);
-  if (!SymlinkPath(socket_target_path, socket_path_) ||
+
+
+  if (
+#ifndef MAC_APP_STORE
+      !SymlinkPath(socket_target_path, socket_path_) ||
+#endif
       !SymlinkPath(cookie, cookie_path_) ||
       !SymlinkPath(cookie, remote_cookie_path)) {
     // We've already locked things, so we can't have lost the startup race,
@@ -992,10 +1015,18 @@ bool ProcessSingleton::Create() {
     return false;
   }
 
+#ifndef MAC_APP_STORE
   SetupSocket(socket_target_path.value(), &sock, &addr);
+#else
+  SetupSocket(socket_path_.value(), &sock, &addr);
+#endif
 
   if (bind(sock, reinterpret_cast<sockaddr*>(&addr), sizeof(addr)) < 0) {
+#ifndef MAC_APP_STORE
     PLOG(ERROR) << "Failed to bind() " << socket_target_path.value();
+#else
+    PLOG(ERROR) << "Failed to bind() " << socket_path_.value();
+#endif
     CloseSocket(sock);
     return false;
   }
diff --git a/chrome/browser/ui/startup/startup_browser_creator_impl.cc b/chrome/browser/ui/startup/startup_browser_creator_impl.cc
index 6ac0bc79..d96ec6a 100644
--- a/chrome/browser/ui/startup/startup_browser_creator_impl.cc
+++ b/chrome/browser/ui/startup/startup_browser_creator_impl.cc
@@ -987,7 +987,7 @@ void StartupBrowserCreatorImpl::InitializeWelcomeRunType(
 #else  // OS_WIN
   // Show the welcome page as the last tab only on first-run.
   if (first_run::ShouldShowWelcomePage())
-    welcome_run_type_ = WelcomeRunType::FIRST_RUN_LAST_TAB;
+    welcome_run_type_ = WelcomeRunType::NONE;
 #endif  // !OS_WIN
 }
 
diff --git a/chrome/common/chrome_constants.cc b/chrome/common/chrome_constants.cc
index a48b5cc..1818ba6 100644
--- a/chrome/common/chrome_constants.cc
+++ b/chrome/common/chrome_constants.cc
@@ -164,7 +164,12 @@ const base::FilePath::CharType kSingletonCookieFilename[] =
     FPL("SingletonCookie");
 const base::FilePath::CharType kSingletonLockFilename[] = FPL("SingletonLock");
 const base::FilePath::CharType kSingletonSocketFilename[] =
+#ifndef MAC_APP_STORE
     FPL("SingletonSocket");
+#else
+    FPL("S");
+#endif
+
 const base::FilePath::CharType kSupervisedUserSettingsFilename[] =
     FPL("Managed Mode Settings");
 const base::FilePath::CharType kThemePackFilename[] = FPL("Cached Theme.pak");
diff --git a/chrome/common/chrome_paths_mac.mm b/chrome/common/chrome_paths_mac.mm
index db32629..0cb0a59 100644
--- a/chrome/common/chrome_paths_mac.mm
+++ b/chrome/common/chrome_paths_mac.mm
@@ -66,7 +66,7 @@ char* ProductDirNameForBundle(NSBundle* chrome_bundle) {
 #if defined(GOOGLE_CHROME_BUILD)
     product_dir_name = "Google/Chrome";
 #else
-    product_dir_name = "Chromium";
+    product_dir_name = "Iridium";
 #endif
   }
 
diff --git a/chrome/common/mac/cfbundle_blocker_unittest.mm b/chrome/common/mac/cfbundle_blocker_unittest.mm
index 72df26a..308bc66 100644
--- a/chrome/common/mac/cfbundle_blocker_unittest.mm
+++ b/chrome/common/mac/cfbundle_blocker_unittest.mm
@@ -27,7 +27,7 @@ TEST(CFBundleBlockerTest, IsBundleAllowed) {
     { nil, nil, false },
 
     // Block bundle IDs that aren't in the whitelist.
-    { @"org.chromium.Chromium.evil", nil, false },
+    { @"de.iridiumbrowser.evil", nil, false },
 
     // The AllowedBundle structure for Google Authetnicator BT doesn't
     // require a version, so this should work equally well with any version
diff --git a/components/os_crypt/keychain_password_mac.mm b/components/os_crypt/keychain_password_mac.mm
index 2ce10cd..76a67ff 100644
--- a/components/os_crypt/keychain_password_mac.mm
+++ b/components/os_crypt/keychain_password_mac.mm
@@ -55,8 +55,8 @@ std::string KeychainPassword::GetPassword() const {
   const std::string service_name = "Chrome Safe Storage";
   const std::string account_name = "Chrome";
 #else
-  const std::string service_name = "Chromium Safe Storage";
-  const std::string account_name = "Chromium";
+  const std::string service_name = "Iridium Safe Storage";
+  const std::string account_name = "Iridium";
 #endif
 
   UInt32 password_length = 0;
diff --git a/content/browser/accessibility/browser_accessibility_cocoa.mm b/content/browser/accessibility/browser_accessibility_cocoa.mm
index 1ae72bd..e5e0f20 100644
--- a/content/browser/accessibility/browser_accessibility_cocoa.mm
+++ b/content/browser/accessibility/browser_accessibility_cocoa.mm
@@ -100,10 +100,12 @@ const int kAXResultsLimitNoLimit = -1;
 
 extern "C" {
 
+#ifndef MAC_APP_STORE
 // See http://openradar.appspot.com/9896491. This SPI has been tested on 10.5,
 // 10.6, and 10.7. It allows accessibility clients to observe events posted on
 // this object.
 void NSAccessibilityUnregisterUniqueIdForUIElement(id element);
+#endif
 
 // The following are private accessibility APIs required for cursor navigation
 // and text selection. VoiceOver started relying on them in Mac OS X 10.11.
@@ -512,7 +514,9 @@ bool InitializeAccessibilityTreeSearch(
 
 - (void)detach {
   if (browserAccessibility_)
+#ifndef MAC_APP_STORE
     NSAccessibilityUnregisterUniqueIdForUIElement(self);
+#endif
   browserAccessibility_ = nullptr;
 }
 
diff --git a/content/browser/renderer_host/render_widget_host_view_mac.mm b/content/browser/renderer_host/render_widget_host_view_mac.mm
index 0581389..bc5f21d 100644
--- a/content/browser/renderer_host/render_widget_host_view_mac.mm
+++ b/content/browser/renderer_host/render_widget_host_view_mac.mm
@@ -2928,9 +2928,12 @@ void RenderWidgetHostViewMac::OnDisplayMetricsChanged(
 // Since this implementation doesn't have to wait any IPC calls, this doesn't
 // make any key-typing jank. --hbono 7/23/09
 //
+
+#ifndef MAC_APP_STORE
 extern "C" {
 extern NSString *NSTextInputReplacementRangeAttributeName;
 }
+#endif
 
 - (NSArray *)validAttributesForMarkedText {
   // This code is just copied from WebKit except renaming variables.
@@ -2939,7 +2942,9 @@ extern NSString *NSTextInputReplacementRangeAttributeName;
         NSUnderlineStyleAttributeName,
         NSUnderlineColorAttributeName,
         NSMarkedClauseSegmentAttributeName,
+#ifndef MAC_APP_STORE
         NSTextInputReplacementRangeAttributeName,
+#endif
         nil]);
   }
   return validAttributesForMarkedText_.get();
diff --git a/content/common/sandbox_mac.mm b/content/common/sandbox_mac.mm
index 6a7f2cf..a10dcca 100644
--- a/content/common/sandbox_mac.mm
+++ b/content/common/sandbox_mac.mm
@@ -490,11 +490,15 @@ bool Sandbox::EnableSandbox(int sandbox_type,
   bool elcap_or_later = base::mac::IsOSElCapitanOrLater();
   if (!compiler.InsertBooleanParam("ELCAP_OR_LATER", elcap_or_later))
     return false;
-
+#ifndef MAC_APP_STORE
   // Initialize sandbox.
   std::string error_str;
   bool success = compiler.CompileAndApplyProfile(&error_str);
   DLOG_IF(FATAL, !success) << "Failed to initialize sandbox: " << error_str;
+#else
+  // We are already sandboxed.
+  bool success = true;
+#endif
   gSandboxIsActive = success;
   return success;
 }
diff --git a/media/capture/device_monitor_mac.mm b/media/capture/device_monitor_mac.mm
index 2ff63e0..5feaad1 100644
--- a/media/capture/device_monitor_mac.mm
+++ b/media/capture/device_monitor_mac.mm
@@ -323,6 +323,7 @@ AVFoundationMonitorImpl::AVFoundationMonitorImpl(
       device_task_runner_(device_task_runner),
       suspend_observer_delegate_(new SuspendObserverDelegate(this)) {
   DCHECK(main_thread_checker_.CalledOnValidThread());
+#ifndef MAC_APP_STORE
   NSNotificationCenter* nc = [NSNotificationCenter defaultCenter];
   device_arrival_ =
       [nc addObserverForName:AVFoundationGlue::
@@ -340,20 +341,25 @@ AVFoundationMonitorImpl::AVFoundationMonitorImpl(
                   usingBlock:^(NSNotification* notification) {
                     OnDeviceChanged();
                   }];
+#endif
   suspend_observer_delegate_->StartObserver(device_task_runner_);
 }
 
 AVFoundationMonitorImpl::~AVFoundationMonitorImpl() {
   DCHECK(main_thread_checker_.CalledOnValidThread());
   suspend_observer_delegate_->ResetDeviceMonitor();
+#ifndef MAC_APP_STORE
   NSNotificationCenter* nc = [NSNotificationCenter defaultCenter];
   [nc removeObserver:device_arrival_];
   [nc removeObserver:device_removal_];
+#endif
 }
 
 void AVFoundationMonitorImpl::OnDeviceChanged() {
   DCHECK(main_thread_checker_.CalledOnValidThread());
+#ifndef MAC_APP_STORE
   suspend_observer_delegate_->OnDeviceChanged(device_task_runner_);
+#endif
 }
 
 }  // namespace
@@ -469,7 +475,9 @@ void DeviceMonitorMac::NotifyDeviceChanged(
     base::SystemMonitor::DeviceType type) {
   DCHECK(thread_checker_.CalledOnValidThread());
   // TODO(xians): Remove the global variable for SystemMonitor.
+#ifndef MAC_APP_STORE
   base::SystemMonitor::Get()->ProcessDevicesChanged(type);
+#endif
 }
 
 }  // namespace media
diff --git a/remoting/branding_Chromium b/remoting/branding_Chromium
index 3c3a43d..df59962 100644
--- a/remoting/branding_Chromium
+++ b/remoting/branding_Chromium
@@ -1,7 +1,7 @@
 APK_PACKAGE_NAME=org.chromium.chromoting
 DAEMON_FILE_NAME=Chromoting Host Service
 IT2ME_HOST_DESCRIPTION=Remote Assistance Host for Chromoting
-MAC_BUNDLE_ID=org.chromium.Chromium
+MAC_BUNDLE_ID=de.iridiumbrowser
 MAC_CREATOR=Cr24
 MAC_HOST_BUNDLE_ID=org.chromium.chromoting.me2me-host
 MAC_HOST_BUNDLE_NAME=ChromotingHost.bundle
diff --git a/ui/shell_dialogs/select_file_dialog_mac.mm b/ui/shell_dialogs/select_file_dialog_mac.mm
index 1ac8122..3d0d5b2 100644
--- a/ui/shell_dialogs/select_file_dialog_mac.mm
+++ b/ui/shell_dialogs/select_file_dialog_mac.mm
@@ -401,7 +401,9 @@ SelectFileDialog* CreateSelectFileDialog(SelectFileDialog::Listener* listener,
         index = 1;
       }
     } else {
+#ifndef MAC_APP_STORE
       CHECK([panel isKindOfClass:[NSOpenPanel class]]);
+#endif
       NSArray* urls = [static_cast<NSOpenPanel*>(panel) URLs];
       for (NSURL* url in urls)
         if ([url isFileURL])
